test/migration_test.dart

```diff
@@ -0,0 +1,391 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+// ignore_for_file: unused_local_variable
+
+import 'dart:async';
+import 'dart:io';
+import 'package:test/test.dart' hide test, throws;
+import '../lib/realm.dart';
+import 'test.dart';
+import '../lib/src/results.dart';
+
+part 'migration_test.g.dart';
+
+@RealmModel()
+@MapTo("Person")
+class _PersonIntName {
+  late int name;
+}
+
+@RealmModel()
+@MapTo('Student')
```

Comment:
Do you think we can extend the "MapTo" API doc so users know it's being used during the migration process?

Reply:
Not sure what you mean - it's not strictly necessary to use `MapTo` here. The reason why I'm doing it is to simulate going from v1 of the app to v2.